File: app/src/main/java/com/elvotra/clean/domain/repository/IPostsRepository.java
package com.elvotra.clean.domain.repository;

import android.support.annotation.NonNull;

import com.elvotra.clean.domain.model.Post;

import java.util.List;

public interface IPostsRepository {

    interface LoadPostsCallback {

        void onPostsLoaded(List<Post> posts);

        void onError(int statusCode);
    }

    interface LoadPostCallback {

        void onPostLoaded(Post post);

        void onError(int statusCode);
    }

    void getPosts(boolean forceUpdate, @NonNull LoadPostsCallback callback);

    void getPost(int postId, @NonNull LoadPostCallback callback);

    void deleteAllData();

    void savePosts(List<Post> posts);
}

File: app/src/main/java/com/elvotra/clean/domain/repository/IPostsRepository.kt
package com.elvotra.clean.domain.repository

import com.elvotra.clean.domain.model.Post
import java.util.ArrayList

interface IPostsRepository {

    interface LoadPostsCallback {

        fun onPostsLoaded(posts: ArrayList<Post>)

        fun onError(statusCode: Int)
    }

    interface LoadPostCallback {

        fun onPostLoaded(post: Post)

        fun onError(statusCode: Int)
    }

    fun getPosts(forceUpdate: Boolean, callback: LoadPostsCallback)

    fun getPost(postId: Int, callback: LoadPostCallback)

    fun deleteAllData()

    fun savePosts(posts: ArrayList<Post>)
}
